src/reformat_libyuv.c

```diff
@@ -663,6 +674,111 @@ avifResult avifImageYUVToRGBLibYUV10bpc(const avifImage * image,
     return AVIF_RESULT_NOT_IMPLEMENTED;
 }
 
+avifResult avifImageYUVToRGBLibYUV12bpc(const avifImage * image,
+                                        avifRGBImage * rgb,
+                                        const struct YuvConstants * matrixYUV,
+                                        const struct YuvConstants * matrixYVU)
+{
+    // See if the current settings can be accomplished with libyuv, and use it (if possible).
+
+    assert((image->depth == 12) && (rgb->depth == 8));
+
+    // libavif uses byte-order when describing pixel formats, such that the R in RGBA is the lowest address,
+    // similar to PNG. libyuv orders in word-order, so libavif's RGBA would be referred to in libyuv as ABGR.
+    // In addition, swapping U and V in any of these calls, along with using the Yvu matrix instead of Yuv matrix,
+    // swaps B and R in these orderings as well. This table summarizes the lookup tables that follow:
+    //
+    // libavif format        libyuv Func     UV matrix (and UV argument ordering)
+    // --------------------  -------------   ------------------------------------
+    // AVIF_RGB_FORMAT_RGB   n/a             n/a
+    // AVIF_RGB_FORMAT_RGBA  *ToARGBMatrix   matrixYVU
+    // AVIF_RGB_FORMAT_ARGB  n/a             n/a
+    // AVIF_RGB_FORMAT_BGR   n/a             n/a
+    // AVIF_RGB_FORMAT_BGRA  *ToARGBMatrix   matrixYUV
+    // AVIF_RGB_FORMAT_ABGR  n/a             n/a
+    // AVIF_RGB_FORMAT_565   n/a             n/a
+
+    avifBool isYVU = lutIsYVU[rgb->format];
+    const struct YuvConstants * matrix = isYVU ? matrixYVU : matrixYUV;
+    int uPlaneIndex = isYVU ? AVIF_CHAN_V : AVIF_CHAN_U;
+    int vPlaneIndex = isYVU ? AVIF_CHAN_U : AVIF_CHAN_V;
+
+    // Lookup table for YUV To RGB Matrix (with filter).
+    typedef int (*YUVToRGBMatrixFilter)(const uint16_t *,
+                                        int,
+                                        const uint16_t *,
+                                        int,
+                                        const uint16_t *,
+                                        int,
+                                        uint8_t *,
+                                        int,
+                                        const struct YuvConstants *,
+                                        int,
+                                        int,
+                                        enum FilterMode);
+    YUVToRGBMatrixFilter lutYuvToRgbMatrixFilter[AVIF_RGB_FORMAT_COUNT][AVIF_PIXEL_FORMAT_COUNT] = {
```

Comment:
given that this is all NULLs. we can omit this for now and add it later when we actually have a non-NULL entry here.

Reply:
It was more convenient to also add a bit depth dimension to the lutYuvToRgbMatrixFilter table, which required a two-dimensional subtable of all NULLs.